### ISSUE TITLE
Password encryption in registry/config.yml for proxy?

### ISSUE THREAD

**Issue by jeffweilee** [goharbor/harbor#4270]
Hello, I am using proxy for remote mirror, so I have to config proxy username and password in ```common/templates/registry/config.yml```. However, the password is in plain code. **Is there any way to have encrypted password rather than plain password in the configuration files for security?**

E.g. 
proxy:
   remoteurl:  [https://remoteurl]
   username: [username]
   password: [**encrypted password rather than plain password**]

I have read https://github.com/vmware/harbor/issues/1329 and https://github.com/vmware/harbor/issues/1329 but still not sure how it supports.

In short, Is there any way to store the password in encrypted format? Or I missed any documents? 
Thank you!

If you are reporting a problem, please make sure the following information are provided:
1)Version of docker engine and docker-compose.
no need.
2)Config files of harbor, you can get them by packaging "harbor.cfg" and files in the same directory, including subdirectory.
no need.
3)Log files, you can get them by package the /var/log/harbor/ .
no need.

**Comment by reasonerjt** [comment]
Strictly speaking as the docker registry's code read plain text from the config.yml I don't think there's a chance to avoid leaking if the docker host is compromised.

**Comment by reasonerjt** [comment]
Closing as won't fix.
